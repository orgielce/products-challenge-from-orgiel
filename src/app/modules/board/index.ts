// export * from './components';
export * from './board.component';
export * from './board.module';
