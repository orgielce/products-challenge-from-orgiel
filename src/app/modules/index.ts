export * from './store.module';
export * from './layout';
