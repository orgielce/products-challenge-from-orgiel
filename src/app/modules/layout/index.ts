// export * from './components';
export * from './layout.component';
export * from './layout.module';
