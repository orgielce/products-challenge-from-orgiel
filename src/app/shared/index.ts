// export * from './directives';
// export * from './enums';
// export * from './filters';
// export * from './locale-l10n';
// export * from './pipes';
// export * from './store';
// export * from './utils';
// export * from './validators';
export * from './guards';
export * from './helpers';
export * from './models';
export * from './routes';
export * from './services';
export * from './store';
