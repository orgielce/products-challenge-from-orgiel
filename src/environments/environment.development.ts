export const environment = {
  barCodeLookupUrl: "https://account.barcodelookup.com/",
};
