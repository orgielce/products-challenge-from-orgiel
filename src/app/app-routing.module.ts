import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

import {PageNotFoundComponent} from "./components";
import {AuthGuard, ROUTES_PATH} from "./shared";

const routes: Routes = [
  {
    path: '',
    loadChildren: () => import('./modules').then(m => m.LayoutModule)
  },
  {
    path: ROUTES_PATH.Login,
    loadChildren: () => import('./modules').then(m => m.AuthModule)
  },
  {
    path: ROUTES_PATH.Products,
    loadChildren: () => import('./modules').then(m => m.BoardModule),
    canActivate: [AuthGuard],
  },
  {
    path: '**', component: PageNotFoundComponent
  },
];

@NgModule({
  imports: [RouterModule.forRoot(routes)],
  exports: [RouterModule]
})
export class AppRoutingModule { }
