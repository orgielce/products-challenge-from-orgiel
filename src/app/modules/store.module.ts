import { NgModule } from '@angular/core';
import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';
import {reducers, metaReducers} from "../shared";

@NgModule({
  imports: [
    StoreModule.forRoot({}, {}),
    EffectsModule.forRoot([]),
    StoreModule.forRoot(reducers, {
      metaReducers
    })
  ]
})
export class NgRxModule { }
