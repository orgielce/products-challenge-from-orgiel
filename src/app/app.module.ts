import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {BrowserAnimationsModule} from "@angular/platform-browser/animations";
import {HttpClientModule} from "@angular/common/http";

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';

import {NgRxModule} from "./modules";
import {PageNotFoundComponent} from "./components";

@NgModule({
  declarations: [
    AppComponent,
    PageNotFoundComponent
  ],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    HttpClientModule,
    NgRxModule,
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule {
}
