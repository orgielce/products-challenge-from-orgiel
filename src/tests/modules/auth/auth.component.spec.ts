import {ComponentFixture, TestBed} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AuthComponent} from "../../../app/modules";
import {CommonModule} from "@angular/common";
import {FormsModule, ReactiveFormsModule} from "@angular/forms";
import {environment} from "../../../environments/environment";

describe('AuthComponent', () => {

  let component: AuthComponent;
  let fixture: ComponentFixture<AuthComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [AuthComponent],
      imports: [
        CommonModule,
        FormsModule,
        ReactiveFormsModule,
        RouterTestingModule
      ],
    })
      .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(AuthComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  test('should create', () => {
    expect(component).toBeTruthy();
  });

  test('should make match with snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });

  test(`should have authForm variable`, () => {
    expect(component.authForm).toBeTruthy();
  });

  test(`should have utils class with invalidField method`, () => {
    const method = component.utils.invalidField(component.authForm, 'key');
    expect(method).toEqual(false);
  });

  test(`should have signUp url environment`, () => {
    const signUpUrl = component.env.barCodeLookupSignUpUrl;
    expect(signUpUrl).toEqual(environment.barCodeLookupSignUpUrl);
  });

  test(`should have login url environment`, () => {
    const loginUrl = component.env.barCodeLookupLoginUrl;
    expect(loginUrl).toEqual(environment.barCodeLookupLoginUrl);
  });

  test('should have dom elements', () => {
    fixture.detectChanges();
    // console.log(compiled.innerHTML);
    const authContainer = compiled.querySelector('.auth-container');
    const authContent = compiled.querySelector('.auth-content');
    const leftColumn = compiled.querySelector('.auth-content-left');
    const rightColumn = compiled.querySelector('form');
    const images = compiled.querySelectorAll('img');
    const labels = compiled.querySelectorAll('label');
    const inputs = compiled.querySelectorAll('input');
    expect( authContainer ).toBeTruthy();
    expect( authContent ).toBeTruthy();
    expect( leftColumn ).toBeTruthy();
    expect( rightColumn ).toBeTruthy();
    expect( inputs.length ).toBe(2);
    expect( labels.length ).toBe(2);
    expect( images.length ).toBe(2);
  });
});
