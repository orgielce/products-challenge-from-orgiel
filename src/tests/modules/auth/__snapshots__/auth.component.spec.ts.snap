// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthComponent should make match with snapshot 1`] = `"<div class="auth-container"><div class="auth-content"><div class="auth-content-left"><div><img src="assets/images/brand.svg" width="100" height="100" alt="SLDMark Brand"><p>we know what you like</p></div><img src="assets/images/brand-slogan-1.png" width="400" height="400" alt="Apps"></div><form novalidate="" ng-reflect-form="[object Object]" class="ng-untouched ng-pristine ng-invalid"><label>Name: </label><input type="text" placeholder="Laura Smitt" name="name" autocomplete="off" formcontrolname="name" ng-reflect-name="name" ng-reflect-ng-class="[object Object]" class="ng-untouched ng-pristine ng-invalid"><label>Key: </label><input type="text" placeholder="Your access key" name="key" autocomplete="off" formcontrolname="key" ng-reflect-name="key" ng-reflect-ng-class="[object Object]" class="ng-untouched ng-pristine ng-invalid"><div>Forget key</div><button type="button" disabled="" ng-reflect-ng-class="[object Object]" class="invalid-btn">Login</button><p>Doesn't have an account yet? <span>SignUp</span></p></form></div></div>"`;
